### PR TITLE
[ClangImporter] Pass -mcx16 to Clang invocation for all x86_64 targets

### PR DESCRIPTION
**Pull request by lorentey**
The Swift runtime and the concurrency module both use -mcx16, so there seems to be little point in not enabling it for Swift code in general.

This allows packages like [swift-atomics](https://github.com/apple/swift-atomics) to rely on double wide atomics always being available. 

(They can't enable this on their own, as `-mcx16` is considered an "unsafe" compiler option; worse, AFAIK there isn't even a way to detect whether cx16 is available from Swift code.  So swift-atomics clients on Linux/x86_64 must manually enable this feature by specifying `-Xcc -mcx16 -Xswiftc -DENABLE_DOUBLEWIDE_ATOMICS` while building their code. This is terrible.)

https://bugs.swift.org/browse/SR-13781

### CODE REVIEW

**Comment by lorentey** [comment]
(Draft because I have no idea what I'm doing, and I haven't even tried building this.) 🙈

**Comment by lorentey** [comment]
@swift-ci test

**Comment by lorentey** [comment]
cc @glessard

**Comment by lorentey** [comment]
@swift-ci test

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -781,6 +781,11 @@ importer::addCommonInvocationArguments(
     invocationArgStrs.push_back("-march=z13");
   }
 
+  if (triple.isOSLinux() && triple.getArch() == llvm::Triple::x86_64) {
```
Comment (compnerd) [inline lib/ClangImporter/ClangImporter.cpp:784]: Any reason to not unconditionally do this?  Does this cause problems on macOS?  I think that we should be doing this on WIndows, Linux, and FreeBSD at least.

Comment (lorentey) [reply lib/ClangImporter/ClangImporter.cpp:784]: Good point; it's already present on Darwin targets, but it doesn't hurt to add it anyway. Updating.

**Comment by lorentey** [comment]
@swift-ci test

**Comment by lorentey** [comment]
```
21:18:26 ******************** TEST 'Swift(linux-x86_64) :: IRGen/cx16.swift' FAILED ********************
21:18:26 Script:
21:18:26 --
21:18:26 : 'RUN: at line 1';   /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/bin/swift-frontend -target x86_64-unknown-linux-gnu  -module-cache-path /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache -swift-version 4  -define-availability 'SwiftStdlib 5.5:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0'  -typo-correction-limit 10  -target x86_64-unknown-linux-gnu -parse-stdlib -disable-objc-interop /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift -module-name main -emit-ir -o - | /usr/bin/python3.7 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/utils/PathSanitizingFileCheck --sanitize BUILD_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64 --sanitize SOURCE_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift --use-filecheck /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/llvm-linux-x86_64/bin/FileCheck  /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift
21:18:26 : 'RUN: at line 2';   /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/bin/swift-frontend -target x86_64-unknown-linux-gnu  -module-cache-path /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache -swift-version 4  -define-availability 'SwiftStdlib 5.5:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0'  -typo-correction-limit 10  -target x86_64-unknown-windows-msvc -parse-stdlib -disable-objc-interop /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift -module-name main -emit-ir -o - | /usr/bin/python3.7 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/utils/PathSanitizingFileCheck --sanitize BUILD_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64 --sanitize SOURCE_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift --use-filecheck /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/llvm-linux-x86_64/bin/FileCheck  /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift
21:18:26 : 'RUN: at line 3';   /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/bin/swift-frontend -target x86_64-unknown-linux-gnu  -module-cache-path /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache -swift-version 4  -define-availability 'SwiftStdlib 5.5:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0'  -typo-correction-limit 10  -target x86_64-unknown-freebsd -parse-stdlib -disable-objc-interop /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift -module-name main -emit-ir -o - | /usr/bin/python3.7 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/utils/PathSanitizingFileCheck --sanitize BUILD_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64 --sanitize SOURCE_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift --use-filecheck /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/llvm-linux-x86_64/bin/FileCheck  /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift
21:18:26 : 'RUN: at line 4';   /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/bin/swift-frontend -target x86_64-unknown-linux-gnu  -module-cache-path /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache -swift-version 4  -define-availability 'SwiftStdlib 5.5:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0'  -typo-correction-limit 10  -target x86_64-apple-macosx10.9 -parse-stdlib -module-name main /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift -emit-ir -o - | /usr/bin/python3.7 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/utils/PathSanitizingFileCheck --sanitize BUILD_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64 --sanitize SOURCE_DIR=/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift --use-filecheck /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/llvm-linux-x86_64/bin/FileCheck  /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift
21:18:26 --
21:18:26 Exit Code: 2
21:18:26 
21:18:26 Command Output (stderr):
21:18:26 --
21:18:26 
<unknown>:0: error: IR generation failure: Cannot read legacy layout file at '/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/swift-linux-x86_64/lib/swift/macosx/layouts-x86_64.yaml'
21:18:26 FileCheck error: '<stdin>' is empty.
21:18:26 FileCheck command line:  /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/buildbot_linux/llvm-linux-x86_64/bin/FileCheck /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/test/IRGen/cx16.swift
21:18:26 
21:18:26 --
```

**Comment by lorentey** [comment]
@swift-ci test

**Review by glessard — APPROVED** [review]
LGTM

**Review by compnerd — APPROVED** [review]

**Pull request merged**